Mathlib/Analysis/Normed/Ring/SmoothingSeminorm.lean

```diff
@@ -0,0 +1,604 @@
+/-
+Copyright (c) 2024 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández
+-/
+import Mathlib.Analysis.Normed.Ring.Seminorm
+import Mathlib.Analysis.SpecialFunctions.Pow.Continuity
+import Mathlib.Topology.MetricSpace.Sequences
+
+/-!
+# smoothingSeminorm
+In this file, we prove [BGR, Proposition 1.3.2/1][bosch-guntzer-remmert] : if `f` is a
+nonarchimedean seminorm on `R`, then `iInf (fun (n : pnat), (f(x^(n : ℕ)))^(1/(n : ℝ)))` is a
+power-multiplicative nonarchimedean seminorm on `R`.
+
+## Main Definitions
+* `smoothingSeminormSeq` : the `ℝ`-valued sequence sending `n` to `(f (x ^ n))^(1/n : ℝ)`.
+* `smoothingSeminorm'` : the iInf of the sequence `f(x^(n : ℕ)))^(1/(n : ℝ)`.
+* `smoothingSeminorm` : if `f 1 ≤ 1` and `f` is nonarchimedean, then `smoothingSeminorm'`
+  is a ring seminorm.
+
+## Main Results
+
+* `tendsto_smoothingSeminorm'` :if `f 1 ≤ 1`, then `smoothingSeminorm' f x` is the limit
+  of `smoothingSeminormSeq f x` as `n` tends to infinity.
+* `isNonarchimedean_smoothingSeminorm'` : if `f 1 ≤ 1` and `f` is nonarchimedean, then
+  `smoothingSeminorm'` is nonarchimedean.
+* `isPowMul_smoothingSeminorm'` : if `f 1 ≤ 1` and `f` is nonarchimedean, then
+  `smoothingSeminorm' f` is power-multiplicative.
+
+## References
+* [S. Bosch, U. Güntzer, R. Remmert, *Non-Archimedean Analysis*][bosch-guntzer-remmert]
+
+## Tags
+
+smoothingSeminorm, seminorm, nonarchimedean
+-/
+
+noncomputable section
+
+open Filter Nat Real
+
+open scoped Topology NNReal
+
+variable {R : Type*} [CommRing R] (f : RingSeminorm R)
+
+section smoothingSeminorm
+
+/-- The `ℝ`-valued sequence sending `n` to `(f (x ^ n))^(1/n : ℝ)`. -/
+def smoothingSeminormSeq (x : R) : ℕ → ℝ := fun n => f (x ^ n) ^ (1 / n : ℝ)
+
+/-- For any positive `ε`, there exists a positive natural number `m` such that
+  `f (x ^ (m : ℕ)) ^ (1 / m : ℝ) < iInf (fun (n : pnat), (f(x ^(n : ℕ)))^(1/(n : ℝ))) + ε/2`. -/
+private theorem smoothingSeminormSeq_hasLimit_m (x : R) {ε : ℝ} (hε : 0 < ε) :
```

Comment:
Do we really want the `_m` at the end of the name?

Reply:
I have renamed this to `smoothingSeminormSeq_exists_pnat`. To be honest, I do not know what would be a good name for this theorem, but since it is private I did not think the naming was too problematic.